### DISCUSSION TITLE
[Feature] Choose mobile folder when downloading files from storage

### DISCUSSION THREAD

**Discussion by AuxiliaryPanther** [immich-app/immich#11896] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, when you upload a file (photo or video) from mobile app to your server, delete it from your device, and then try to download it from your server, it will put the image or video into your mobile device storage in the Pictures or Videos folders, respectively.

It would be fantastic to be able to choose in settings where files get downloaded to. I am using the Android version. 

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by zlnvch** [comment]
Actually, on the current Android app, in my testing, images get saved to a an immich folder. However on iOS they do not go into an immich album, but are placed in the main Recents category.

It would be nice if it could do this on iOS as well. Currently, for example, if I save an image from partner sharing onto my iOS device, as there is no way to exclude an immich album from sync, next time I run the backup it will backup the image to the server, thus keeping two copies of an identical image on the server.

On the other hand, I suppose there are also use cases where you wish to restore your own images on immich to your device back into the albums they were originally in when you uploaded them.

I suppose putting them back in the original album folder, can be problematic on Android because while on iOS, images can actually be in multiple albums, on Android it can only be in one folder. So if you've added an image to multiple albums on immich web, it would have to choose one to download to. I'm not sure what would be the best way of picking which album.

To summarize, I don't think it's necessary to complicate the UI with an option. I think this is what we need, and I would split this into two separate tasks, so that we can get the low hanging fruit one out of the way quickly.

1) Putting all images/videos downloaded from immich into an immich folder/album. This is already working on Android, it needs to be done on iOS.
2) Putting images download from your own immich account (since only for non-shared images you get album info) into the correct album back on your phone. This leads to some questions for Android, which doesn't support multi-album, but should be simple on iOS.

**Reply by NicholasFlamy** [reply]
> Actually, on the current Android app, in my testing, images get saved to a an immich folder.

This is relatively new and a very welcome change.



> To summarize, I don't think it's necessary to complicate the UI with an option. I think this is what we need, and I would split this into two separate tasks, so that we can get the low hanging fruit one out of the way quickly.
> 
> 1. Putting all images/videos downloaded from immich into an immich folder/album. This is already working on Android, it needs to be done on iOS.
> 2. Putting images download from your own immich account (since only for non-shared images you get album info) into the correct album back on your phone. This leads to some questions for Android, which doesn't support multi-album, but should be simple on iOS.

I believe the original post was focused on setting a location to download stuff to (including a folder on a microSD card).

**Comment by serfriz** [comment]
Any update on this? In my Pixel 8 they get downloaded to the gallery, so if I download a picture shared by my partner, it gets re-uploaded again in my account.

**Comment by fakkoweb** [comment]
This would be also useful to find that content into apps like Google Photo, which have ability to advance edit such photos.
By downloading them on the default folder (DCIM/immich) they just get mixed with all other photos/videos made.

**Comment by SinTan1729** [comment]
The current behavior on Android seems to be that the files are stored inside `DCIM/Immich`. It'll be really nice to have an option to move it somewhere else e.g. `Downloads/Immich` as it'd make much more sense for many who backup their DCIM directories, or people like me who use external libraries with externally synced DCIM and don't want Immich using the DCIM directory and creating duplicate entries.

**Reply by NicholasFlamy** [reply]
> backup their DCIM directories

It will show up as a separate entry in the Immich app. Other apps should be able to exclude that in some way.


> external libraries with externally synced DCIM

If you're using Syncthing or something, you can exclude the directory.

**Comment by qkrrmsdud1** [comment]
Google Photos ignores subdirectories in the DCIM folder, so `DCIM/Camera` and `DCIM/Immich` both show up in the Camera album in Google Photos, which makes it a lot more difficult to find assets downloaded from Immich in Google Photos. It would be nice to be able to download from Immich to a folder like `Pictures/Immich`. I think the best option would just be to allow the user to choose any folder, though.

**Reply by NicholasFlamy** [reply]
While I do think Pictures makes more sense, frick Google Photos, they should do it correctly.

**Comment by denny-cube** [comment]
Actually imho the best would be that if an album is synced with a local (Android for me) folder, if I download a picture from that album it gets saved in that related folder.
Of course others could have different preferences, so the best should be to have some settings in the app to define the behaviour.

**Reply by dalaranunion** [reply]
Indeed, this could also utilise the state of the file (e.g. in cloud, not in cloud, saved in both)

Also presence of the file should prevent file re-download. At the moment hitting the "Download" will create duplicates of the same file.

**Reply by TheRestRat** [reply]
Exactly what I would find logical by design. This could be a choice ofc, but would be quite rational

**Comment by NicholasFlamy** [comment]
Something related to this I've run into is that downloading images from a shared album or partner shareing results in Immich uploading it to my library.

**Comment by PlasmPlayer** [comment]
With the OneDrive integration ending with Samsung gallery, this feature is needed now more than ever to help make it a seamless transition to using Immich full time

**Comment by itsdapi** [comment]
iOS user needs this feature!!
